nixos/lib/systemd-lib.nix

```diff
@@ -396,6 +396,13 @@ in rec {
       text = commonUnitText def +
         ''
           [Service]
+          ${let env = def.environment;
+            in concatMapStrings (n:
+              let s = optionalString (env.${n} != null)
+                "Environment=${builtins.toJSON "${n}=${env.${n}}"}\n";
```

Comment:
Cheeky.
```suggestion
                "Environment=${''"${n}=${env.${n}}"''}\n";
```

Reply:
This is how it was done in the regular service version of this. For instance, what if your `env.${n}` has a `\n` character in it? With `toJSON` it'll get properly encoded, whereas that would corrupt the unit file.

Reply:
Ah, didn't think of that.